### PR TITLE
Qb 573 REST API for register and pot modules

### PR DESCRIPTION
**Pull request by hong-pang**
Added query APIs:
1. /register/staking
2. /register/params
3. /register/resource-nodes?page=1&limit=2
4. /register/indexing-nodes?page=1&limit=2
5. /pot/report/epoch/{epoch}
6. /pot/rewards/epoch/1?owner={NodeWalletAddress}&page=1&limit=2
7. /register/staking/address/{nodeAddress}
8. /register/staking/owner/{nodeWalletAddress}?page=1&limit=6

### CODE REVIEW

File: x/register/keeper/querier.go [hunk 1]
```diff
@@ -78,3 +101,357 @@ func GetNetworkSet(ctx sdk.Context, k Keeper) ([]byte, error) {
...
+		if params.OwnerAddr.Empty() || n.OwnerAddress.Equals(params.OwnerAddr) {
+			filteredNodes = append(filteredNodes, n)
+		}
+	}
+
+	filteredNodes = k.resPagination(filteredNodes, params)
+	return filteredNodes
+}
+
+func (k Keeper) resPagination(filteredNodes []types.ResourceNode, params QueryNodesParams) []types.ResourceNode {
```
Comment (alexstratos) [inline x/register/keeper/querier.go:419]: give full name of resource node

File: x/register/keeper/querier.go [hunk 2]
```diff
@@ -78,3 +101,357 @@ func GetNetworkSet(ctx sdk.Context, k Keeper) ([]byte, error) {
...
+	start, end := client.Paginate(len(filteredNodes), params.Page, params.Limit, QueryDefaultLimit)
+	if start < 0 || end < 0 {
+		filteredNodes = nil
+	} else {
+		filteredNodes = filteredNodes[start:end]
+	}
+	return filteredNodes
+}
+
+func (k Keeper) indPagination(filteredNodes []types.IndexingNode, params QueryNodesParams) []types.IndexingNode {
```
Comment (alexstratos) [inline x/register/keeper/querier.go:429]: give full name of index node

**Review by alexstratos — APPROVED** [review]

File: x/register/keeper/querier.go (hunk 1 above)
Comment (hong-pang) [reply x/register/keeper/querier.go]: rename function resPagination to resourceNodesPagination

File: x/register/keeper/querier.go (hunk 2 above)
Comment (hong-pang) [reply x/register/keeper/querier.go]: rename function indPagination to indexingNodesPagination

**Pull request merged**